llvm/include/llvm/Support/ErrorHandling.h

```diff
@@ -61,11 +61,14 @@ struct ScopedFatalErrorHandler {
 /// @deprecated Use reportFatalInternalError() or reportFatalUsageError()
 /// instead.
 [[noreturn]] LLVM_ABI void report_fatal_error(const char *reason,
-                                              bool gen_crash_diag = true);
+                                              bool gen_crash_diag = true,
+                                              bool is_usage_error = false);
 [[noreturn]] LLVM_ABI void report_fatal_error(StringRef reason,
-                                              bool gen_crash_diag = true);
+                                              bool gen_crash_diag = true,
+                                              bool is_usage_error = false);
 [[noreturn]] LLVM_ABI void report_fatal_error(const Twine &reason,
-                                              bool gen_crash_diag = true);
+                                              bool gen_crash_diag = true,
+                                              bool is_usage_error = false);
```

Comment:
These changes don't make sense: gen_crash_diag *is* already the flag that indicates whether this is a usage error and controls whether a stack trace is displayed.

Calling reportFatalUsageError() will correctly not display a stack trace and not ask for a bug report on Linux. If this is not the case on Windows, you need to identify why there is a difference in behavior. Adding an extra flag here is certainly the wrong thing to do though.

Reply:
Thanks. I thought that adding a new flag would mean that the behaviours of 1. `report_fatal_error` calling `abort()` and 2. The signal handler having usage or internal error behaviour were independent - which might be required. I'll update to just use gen_crash_diag - that will make for a smaller patch anyway.

As to why Windows has different behaviour - on Linux LLVM once called the signal handler for `report_fatal_error` IIUC but this seems to have changed in: https://github.com/llvm/llvm-project/commit/288999b82959d6a057bfd478a196961215baf792.

Reply:
@nikic I'm just using `gen_crash_diag` now. Any good?

Reply:
> As to why Windows has different behaviour - on Linux LLVM once called the signal handler for `report_fatal_error` IIUC but this seems to have changed in: [288999b](https://github.com/llvm/llvm-project/commit/288999b82959d6a057bfd478a196961215baf792).

So, can we do the same change for Windows? Basically pass false to Cleanup in the Windows RunInterruptHandlers, or something along those lines?

Reply:
@nikic I have just realised that I might be able to be a bit braver and not pass anything into the signal handlers but instead just not call them in Windows `signal.inc`. Let me try that...

Reply:
> > As to why Windows has different behaviour - on Linux LLVM once called the signal handler for `report_fatal_error` IIUC but this seems to have changed in: [288999b](https://github.com/llvm/llvm-project/commit/288999b82959d6a057bfd478a196961215baf792).
> 
> So, can we do the same change for Windows? Basically pass false to Cleanup in the Windows RunInterruptHandlers, or something along those lines?

Seems to work and much better. Thanks!

Reply:
My remaining question here would be whether we actually need the ExecuteSignalHandlers argument on RunInterruptHandlers, or can always assume it is false like on Linux. If I understood things correctly, the way this is supposed to work is that if GenCrashDiag is true then the following abort() is going to run the signal handlers anyway?

Reply:
Good questions. 

>If I understood things correctly, the way this is supposed to work is that if GenCrashDiag is true then the following abort() is going to run the signal handlers anyway?

Yes. To keep the previous behaviour where the signal handlers are always run. I need to poke at it a bit. Perhaps the output is OK on Windows without running the signal handlers. I don't understand why the difference exists between the two platforms.